### PR TITLE
XPath Parser Error Message Fix

### PR DESCRIPTION
**Pull request by ctsims**
Fixes an issue where an error message would always display "null" as part of its message, rather than a useful piece of context.

### CODE REVIEW

File: src/main/java/org/javarosa/xform/parse/XFormParser.java [hunk 1]
```diff
@@ -1616,7 +1616,7 @@ public static XPathReference getAbsRef(XPathReference ref, TreeReference parentR
 
         tref = tref.parent(parentRef);
         if (tref == null) {
-            throw new XFormParseException("Binding path [" + tref + "] not allowed with parent binding of [" + parentRef + "]");
+            throw new XFormParseException("Binding path [" + ref.getReference().toString(true) + "] not allowed with parent binding of [" + parentRef + "]");
```
Comment (wpride) [inline src/main/java/org/javarosa/xform/parse/XFormParser.java]: Looks from the conditional block above like `ref` can be null - should we be handling that here?

Comment (ctsims) [inline src/main/java/org/javarosa/xform/parse/XFormParser.java]: Oh, I was reading that output wrong, good catch. Will address.

**Comment by ctsims** [comment]
@wpride thanks for that, addressed.

File: src/main/java/org/javarosa/xform/parse/XFormParser.java [hunk 2]
```diff
@@ -1614,9 +1614,10 @@ public static XPathReference getAbsRef(XPathReference ref, TreeReference parentR
             tref = TreeReference.selfRef(); //only happens for <group>s with no binding
         }
 
+        TreeReference refPreContextualization = tref;
```
Comment (amstone326) [inline src/main/java/org/javarosa/xform/parse/XFormParser.java]: @ctsims Assuming I'm missing something, but doesn't this just make `refPreContextualization` have a pointer to `tref`? How will this value remain different?

Comment (ctsims) [inline src/main/java/org/javarosa/xform/parse/XFormParser.java]: tref gets redefined below which is what ends up making its value null if the contextualization fails. This just retains it's "pre-contextualized" state, which is generally a relative reference  like ("../something") which didn't work

**Pull request merged**